src/model/resolvable.ts

```diff
@@ -18,9 +18,13 @@ import {Feature, ScannedFeature} from './feature';
 /**
  * A ScannedFeature that needs to resolve other Features to build its final
  * representation.
+ *
+ * TODO(aomarks) Is it right that I made this an abstract class? ScannedFeature
```

Comment:
I think Resolvable must extend ScannedFeature, as it doesn't make sense to resolve anything but a scanned feature.

We have resolve methods on some non-features (like references), but they're deliberately called on instances of known specific types so no interfaces are needed there. We may want to add arguments to those resolve method calls too.

I'm kinda in favor of extending from ScannedFeature and having folks pass stuff up through the constructor, as a way to reduce duplication of properties and better ensure that constraints are met. e.g. we could ensure that sourceRange is always set.